core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -224,9 +253,9 @@ boolean checkRequestParameters(final RestRequest request, final RestChannel chan
         if (request.paramAsBoolean("error_trace", false) && channel.detailedErrorsEnabled() == false) {
             try {
                 XContentBuilder builder = channel.newErrorBuilder();
-                builder.startObject().field("error", "error traces in responses are disabled.").endObject().string();
+                builder.startObject().field("error", "error traces in responses are disabled.").endObject();
                 RestResponse response = new BytesRestResponse(BAD_REQUEST, builder);
-                response.addHeader("Content-Type", "application/json");
+                response.addHeader("Content-Type", builder.contentType().mediaType());
```

Comment:
this could have meant that we sent back the wrong content type if for some reason we didn't send JSON back. In fact I will remove this line since this header is added by default.